dojson/utils.py

```diff
@@ -45,11 +45,11 @@ def wrapper(self, key, value, **kwargs):
 
 
 def filter_values(f):
-    """Remove None values from dictionary."""
+    """Remove None or empty values from dictionary."""
     @functools.wraps(f)
     def wrapper(*args, **kwargs):
         out = f(*args, **kwargs)
-        return dict((k, v) for k, v in six.iteritems(out) if v is not None)
+        return dict((k, v) for k, v in six.iteritems(out) if v or v is False or v == 0)
```

Comment:
Without explicitly listing False and 0, It would not filter away empty dict, set and list as per the proposal.